### PR TITLE
0220 delete non prod code in emqx_resource.erl

### PR DESCRIPTION
**Pull request by zmstone**
Delete unused code.

### CODE REVIEW

**Review by qzhuyan — DISMISSED** [review]

File: apps/emqx_resource/src/proto/emqx_resource_proto_v2.erl [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+%%--------------------------------------------------------------------
+
+-module(emqx_resource_proto_v2).
+
+-behaviour(emqx_bpapi).
+
+-export([
+    introduced_in/0,
+    reset_metrics/1
+]).
```
Comment (thalesmg) [inline apps/emqx_resource/src/proto/emqx_resource_proto_v2.erl:24]: Q: shouldn't this have the other non-deprecated functions as well (i.e., `recreate` and `remove`)?

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_resource/src/proto/emqx_resource_proto_v2.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx_resource/src/proto/emqx_resource_proto_v2.erl:24]: I don't think so.
The older version nodes will call the `v1` module (which is not deleted yet) during rolling upgrade.

Comment (zmstone) [reply apps/emqx_resource/src/proto/emqx_resource_proto_v2.erl:24]: ah, forgot to add comments.

**Review by thalesmg — APPROVED** [review]

File: apps/emqx_resource/src/proto/emqx_resource_proto_v2.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx_resource/src/proto/emqx_resource_proto_v2.erl:24]: Deleted the "deprecated" comments.
We need the v1 module to stick around to pin the `_local` flavor APIs for xref and dialyzer checks.

**Pull request merged**